ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -45,13 +56,25 @@ type E_GO_BACK = {
   type: "GO_BACK";
 };
 
+type E_ADD_SELF_CONSENT = {
+  type: "ADD_SELF_CONSENT";
+  data: SelfConsentDTO;
+  page: number;
+};
+type E_ALL_CRITERIA_ACCEPTED = {
+  type: "ALL_CRITERIA_ACCEPTED";
+};
+
 type Events =
   | E_SELECT_INITIATIVE
   | E_ACCEPT_TOS
   | E_ACCEPT_REQUIRED_PDND_CRITERIA
   | E_ACCEPT_REQUIRED_SELF_CRITERIA
   | E_QUIT_ONBOARDING
-  | E_GO_BACK;
+  | E_GO_BACK
+  | E_ADD_SELF_CONSENT
+  | E_ACCEPT_REQUIRED_BOOL_CRITERIA
+  | E_ALL_CRITERIA_ACCEPTED;
```

Comment:
```suggestion
  | E_ALL_CRITERIA_ACCEPTED;
```

Avoid naming an event with the same name of a state. This can be confusing.